app/models/piece.rb

```diff
@@ -1,3 +1,27 @@
 class Piece < ApplicationRecord
   belongs_to :game
-end
+
+
+  def is_obstructed?(destination)
+
+    x1 = cell_x #lets us know where the piece starts
+    y1 = cell_y
+    x2 = destination.first 
+    y2 = destination.last
+    del_x = (x2 - x1).positive? ? 1 : -1
+    del_y = (y2 - y1).positive? ? 1 : -1
+    dest_x = x2 - del_x 
+    dest_y = y2 - del_y 
+
+
+    while x1 != dest_x && y1 != dest_y 
+      x1 = del_x + x1 unless x1 == dest_x
+      y1 = del_y + y1 unless y1 == dest_y
+      return true if Piece.find_by(cell_x: x1, cell_y: y1, game_id: game_id).present?
```

Comment:
cool way to do it! looks like most cases are covered because of the way you subtracted off delta x or y for destination x or y. Definitely good to pull into master, just some stuff to think about to add onto what you already built. What about a knight that can jump pieces of the same type? or how will we tell the difference between an obstruction, ie a white rook being blocked by a white pawn, and and piece being taken, ie a white rook taking a black pawn ---- i think we will have to add piece ID in here but I am not sure where yet....